compiler/grainformat/reformat.re

```diff
@@ -3308,258 +4127,177 @@ let print_primitive_value_description =
-
-  // check to see if we have a comment to disable formatting
-
-  let disable_formatting =
-    List.exists(is_disable_formatting_comment, leading_comments);
-
-  Walktree.remove_used_comments(leading_comments, []);
-
-  // remove all nodes before this top level statement
-
-  Walktree.remove_nodes_before(data.ptop_loc);
-
-  let (stmt_leading_comment_docs, prevLine) =
-    print_leading_comments(leading_comments, previous_line);
-
-  let stmt_leading_comment_docs =
-    if (List.length(leading_comments) > 0) {
-      stmt_leading_comment_docs;
-    } else {
-      Doc.nil;
-    };
-
-  let blank_line_above =
-    if (get_loc_line(data.ptop_loc) > prevLine + 1) {
-      Doc.hardLine;
-    } else {
-      Doc.nil;
-    };
-  let attribute_text = print_attributes(attributes);
-
-  if (disable_formatting) {
-    let originalCode = get_original_code(data.ptop_loc, original_source);
-    // need to remove any comments that were inside the disabled block
-
-    Walktree.remove_comments_in_ignore_block(data.ptop_loc);
-
-    Doc.concat([
-      stmt_leading_comment_docs,
-      blank_line_above,
-      attribute_text,
-      Doc.group(Doc.text(originalCode)),
-    ]);
-  } else {
-    let without_comments =
-      switch (data.ptop_desc) {
-      | PTopImport(import_declaration) => import_print(import_declaration)
-      | PTopForeign(export_flag, value_description) =>
-        let export =
-          switch (export_flag) {
-          | Nonexported => Doc.text("import ")
-          | Exported => Doc.text("export ")
-          };
-        Doc.concat([
-          export,
-          Doc.text("foreign wasm "),
-          print_foreign_value_description(value_description, original_source),
-        ]);
-      | PTopPrimitive(export_flag, value_description) =>
-        let export =
-          switch (export_flag) {
-          | Nonexported => Doc.nil
-          | Exported => Doc.text("export ")
-          };
-        Doc.concat([
-          export,
-          Doc.text("primitive "),
-          print_primitive_value_description(
-            value_description,
-            original_source,
-          ),
-        ]);
-      | PTopData(data_declarations) =>
-        data_print(data_declarations, original_source)
-      | PTopLet(export_flag, rec_flag, mut_flag, value_bindings) =>
-        print_value_bind(
-          export_flag,
-          rec_flag,
-          mut_flag,
-          value_bindings,
-          data.ptop_loc,
-          original_source,
-        )
-      | PTopExpr(expression) =>
-        print_expression(
-          ~parentIsArrow=false,
-          ~endChar=None,
+  let without_comments =
+    switch (data.ptop_desc) {
+    | PTopImport(import_declaration) =>
+      import_print(~comments, ~original_source, import_declaration)
+    | PTopForeign(export_flag, value_description) =>
+      let export =
+        switch (export_flag) {
+        | Nonexported => Doc.text("import ")
+        | Exported => Doc.text("export ")
```

Comment:
Wait, what?

Reply:
Needed for this:
`import foreign wasm args_get: (WasmI32, WasmI32) -> WasmI32 from "wasi_snapshot_preview1"`

Reply:
Sorry, I totally didn't know we had an `export foreign` syntax, but grepping for it makes sense why we need/have it. Haha. 🙊